adwords_client/adwordsapi/batch_job_service.py

```diff
@@ -103,6 +103,7 @@ def prepare_job(self, client_customer_id=None):
         self.prepare_mutate()
         self.helper.add_batch_job_operation('ADD')
         self.batch_job = self.mutate(client_customer_id)
+        logger.debug('Created new batchjob:\n{}'.format(str(self.batch_job)))
         self.helper = BatchJobHelper(self)
```

Comment:
- Nos loggers, há interpolação automática para `%s`-style entre os parâmetros posicionais
(mas não para `.format`-style):
  ```python
  >>> logging.warning('foo %s', 3.14000)
  WARNING:root:foo 3.14
  ```
Caso o loglevel tenha filtrado essa chamada então não há o custo de executar a interpolação e consequente conversão. O módulo de `logging` faz essa otimização automaticamente. Do jeito atual sempre há o custo de conversão mesmo que o log não vá para lugar algum.

- Em `.format()`-style, vc pode internamente forçar a conversão p/ `str`, `repr` ou outro tipo:
   ```python
   >>> '{}'.format(object.__dict__)   # Defaults to !s (__str__)
   "{'__repr__': <slot wrapper ..."
   >>> '{!r}'.format(object.__dict__)   # Force to !r (__repr__)
   "mappingproxy({'__repr__': <slot wrapper ..."
   >>> '{!s}'.format(object.__dict__)   # Force to !s (__str__)
   "{'__repr__': <slot wrapper ..."

**Sugiro usar uma dessas duas abordagens, ao invés de converter para string via `str(self.batch_job)`**